orc8r/cloud/go/services/certifier/obsidian/handlers/handlers.go

```diff
@@ -0,0 +1,223 @@
+		Username: username,
+		Password: password,
+	}
+	err := certifier.CreateUser(c.Request().Context(), user)
+	return err
+}
+
+func getUserHandler(c echo.Context) error {
+	username := c.Param("username")
+	user, err := certifier.GetUser(c.Request().Context(), username)
+	if err != nil {
+		return echo.NewHTTPError(http.StatusInternalServerError, err)
+	}
+	return c.JSON(http.StatusOK, user)
+}
+
+func updateUserHandler(c echo.Context) error {
+	username := c.Param("username")
+	var updatePassword string
+	err := json.NewDecoder(c.Request().Body).Decode(&updatePassword)
+	if err != nil {
+		return echo.NewHTTPError(http.StatusInternalServerError, fmt.Errorf("error decoding request body for updating user: %v", err))
+	}
+	newUser := &protos.User{Username: username, Password: []byte(updatePassword)}
+	certifier.UpdateUser(c.Request().Context(), newUser)
+	if err != nil {
+		return echo.NewHTTPError(http.StatusInternalServerError, err)
+	}
+	return nil
+}
+
+func deleteUserHandler(c echo.Context) error {
+	username := c.Param("username")
+	deleteUser := &protos.User{Username: username}
+	err := certifier.DeleteUser(c.Request().Context(), deleteUser)
+	if err != nil {
+		return echo.NewHTTPError(http.StatusInternalServerError, fmt.Errorf("error deleting user: %v", err))
+	}
+	return nil
+}
+
+func getUserTokensHandler(c echo.Context) error {
+	username := c.Param("username")
+	res, err := certifier.ListUserTokens(c.Request().Context(), &protos.User{Username: username})
+	if err != nil {
+		return echo.NewHTTPError(http.StatusInternalServerError, fmt.Errorf("failed to list user tokens: %v", err))
+	}
+	return c.JSON(http.StatusOK, res)
+}
+
+func addUserTokenHandler(c echo.Context) error {
+	username := c.Param("username")
+
+	resources := &models.Resources{}
+	if err := c.Bind(resources); err != nil {
+		return echo.NewHTTPError(http.StatusBadRequest, err)
+	}
+	if err := resources.Validate(strfmt.Default); err != nil {
+		return echo.NewHTTPError(http.StatusBadRequest, err)
+	}
+
+	resourceList := resourcesModelToProto(resources)
+	req := &protos.AddUserTokenRequest{
+		Username:  username,
+		Resources: resourceList,
+	}
+	err := certifier.AddUserToken(c.Request().Context(), req)
+	return err
+}
+
+func deleteUserTokenHandler(c echo.Context) error {
+	username := c.Param("username")
+	token := c.Param("token")
+	req := &protos.DeleteUserTokenRequest{
+		Username: username,
+		Token:    token,
+	}
+	err := certifier.DeleteUserToken(c.Request().Context(), req)
+	return err
+}
+
+func loginHandler(c echo.Context) error {
+	data := &models.User{}
+	if err := c.Bind(data); err != nil {
+		return echo.NewHTTPError(http.StatusBadRequest, err)
+	}
+	if err := data.Validate(strfmt.Default); err != nil {
+		return echo.NewHTTPError(http.StatusBadRequest, err)
+	}
+	username := fmt.Sprintf("%v", *data.Username)
+	password := []byte(fmt.Sprintf("%v", *data.Password))
+	user := &protos.User{
+		Username: username,
+		Password: password,
+	}
```

Comment:
1. Make the swagger username/password non-nil (since it's required), then you won't have to deal with ptrs
2. Make the proto password string, will be simpler (unless you had strong reason to make it bytes -- lmk if so)

Reply:
1. It's already required... 🙃
2. Not a super strong reason but needed it to be bytes when using encrypting the password and either way I would have to convert things somewhere ig.